icicle-compiler/test/Icicle/Test/Sea/Zebra.hs

```diff
@@ -402,23 +391,29 @@ schemaOfType ty = case ty of
         Schema.Variant (Schema.VariantName "left")  <$> schemaOfType x
       rightOf x =
         Schema.Variant (Schema.VariantName "right") <$> schemaOfType x
-    in Schema.Enum <$> leftOf a <*> (Boxed.singleton <$> rightOf b)
+    in do
+      a' <- leftOf a
+      b' <- rightOf b
+      pure . Schema.Enum . Cons.fromNonEmpty $
+        a' :| [b']
 
   MapT k v -> do
-    k' <- schemaOfType' k
-    v' <- schemaOfType' v
-    pure . Schema.Struct . Boxed.fromList $
-      [ Schema.Field (Schema.FieldName "keys") (Schema.Array k')
-      , Schema.Field (Schema.FieldName "vals") (Schema.Array v') ]
+    k' <- schemaOfType k
+    v' <- schemaOfType v
+    pure . Schema.Nested . Schema.Array . Schema.Struct . Cons.fromNonEmpty $
+           Schema.Field (Schema.FieldName "key") k'
+      :| [ Schema.Field (Schema.FieldName "val") v' ]
```

Comment:
yeah, I think we might as well, when it comes to outputting zebra, writing out real maps is what we want I think